src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/DetentionFacilityEditAppealHandler.java

```
@@ -63,31 +60,31 @@ public PreSubmitCallbackResponse<AsylumCase> handle(
-                asylumCase.clear(IRC_NAME);
-                asylumCase.clear(OTHER_DETENTION_FACILITY_NAME);
-            }
-
-            //Clear all 'irc' & 'prison' fields
-            if (facility.equals(OTHER.getValue())) {
-                log.info("Clearing IRC and Prison details for Other Detention Centre");
-                asylumCase.clear(IRC_NAME);
-                asylumCase.clear(PRISON_NAME);
-                asylumCase.clear(PRISON_NOMS);
-                asylumCase.clear(CUSTODIAL_SENTENCE);
-                asylumCase.clear(HAS_PENDING_BAIL_APPLICATIONS);
-                asylumCase.clear(BAIL_APPLICATION_NUMBER);
+            switch (DetentionFacility.from(facility)) {
+                case IRC:
+                    //Clear all 'prison' & 'other' fields
+                    log.info("Clearing Prison details for an IRC Detention centre.");
+                    asylumCase.clear(PRISON_NAME);
+                    asylumCase.clear(PRISON_NOMS);
+                    asylumCase.clear(OTHER_DETENTION_FACILITY_NAME);
+                    asylumCase.clear(CUSTODIAL_SENTENCE);
+                    break;
+                case PRISON:
+                    //Clear all 'irc' & 'other' fields
+                    log.info("Clearing IRC details for a Prison Detention Centre");
+                    asylumCase.clear(IRC_NAME);
+                    asylumCase.clear(OTHER_DETENTION_FACILITY_NAME);
+                    break;
+                case OTHER:
+                    //Clear all 'irc' & 'prison' fields
+                    log.info("Clearing IRC and Prison details for Other Detention Centre");
+                    asylumCase.clear(IRC_NAME);
+                    asylumCase.clear(PRISON_NAME);
+                    asylumCase.clear(PRISON_NOMS);
+                    asylumCase.clear(CUSTODIAL_SENTENCE);
+                    break;
+                default:
+                    // should never execute, DetentionFacility.from() throws error
```

can you throw an error here instead of comment?